### ISSUE TITLE
Data validation check missing from RealmBaseAdapter

### ISSUE THREAD

**Issue by Zhuinden** [realm/realm-android-adapters#112]
https://github.com/realm/realm-android-adapters/blob/master/adapters/src/main/java/io/realm/RealmRecyclerViewAdapter.java#L191-L193

the validation check in `RealmRecyclerViewAdapter`:

``` java
    private boolean isDataValid() {
        return adapterData != null && adapterData.isValid();
    }
```

This is missing from `RealmBaseAdapter`:

``` java
// RecyclerViewAdapter
    @Override
    public int getItemCount() {
        //noinspection ConstantConditions
        return isDataValid() ? adapterData.size() : 0;
    }

// BaseAdapter
    @Override
    public int getCount() {
        if (adapterData == null) {
            return 0;
        }
        return adapterData.size();
    }

```

#### Version of Realm and tooling
Realm version(s): 3.1.4

Android Studio version: -

Which Android version and device: -

**Comment by zaki50** [comment]
@Zhuinden thanks.
I made a PR.

**Comment by beeender** [comment]
fixed by #117
